safesocket/safesocket_darwin.go

```diff
@@ -4,157 +4,164 @@
 package safesocket
 
 import (
-	"bufio"
-	"bytes"
-	"errors"
+	crand "crypto/rand"
 	"fmt"
+	"log"
 	"net"
-	"os"
-	"os/exec"
-	"path/filepath"
-	"strconv"
-	"strings"
 	"sync"
-	"time"
-
-	"tailscale.com/version"
 )
 
+var SameUserProofTokenLength = 10
+
 func init() {
-	localTCPPortAndToken = localTCPPortAndTokenDarwin
+	localTCPPortAndToken = LocalTCPPortAndTokenDarwin
+}
+
+type darwinCredentials struct {
+	mu    sync.Mutex
+	creds struct {
+		token string
+		port  int
+	}
+}
+
+var dc darwinCredentials
+
+func LocalTCPPortAndTokenDarwin() (int, string, error) {
```

Comment:
exported, so needs docs.  And naming those result parameters (at least the first two) would help with such a unique signature.